api/org/commcare/api/persistence/SqlIndexedStorageUtility.java

```diff
@@ -0,0 +1,388 @@
+        } catch (SQLException | ClassNotFoundException e) {
+            e.printStackTrace();
+        }
+        return new JdbcSqlStorageIterator<T>(preparedStatement, resultSet, this.getNumRecords(), this);
+    }
+
+    @Override
+    public int getTotalSize() {
+        //serialize and test blah blah.
+        return 0;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return this.getNumRecords() <= 0;
+    }
+
+    @Override
+    public byte[] readBytes(int id) {
+        PreparedStatement preparedStatement = null;
+        Connection c = null;
+        try {
+            c = getConnection();
+            preparedStatement = SqlHelper.prepareIdSelectStatement(c, this.tableName, id);
+            if (preparedStatement == null) {
+                return null;
+            }
+            ResultSet rs = preparedStatement.executeQuery();
+            return rs.getBytes(org.commcare.modern.database.DatabaseHelper.DATA_COL);
+        } catch (Exception e) {
+            System.out.println("SqlIndexedStorageUtility readBytes exception: " + e);
+        } finally{
+            try {
+                if (preparedStatement != null) {
+                    preparedStatement.close();
+                }
+                if (c != null) {
+                    c.close();
+                }
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public void update(int id, Persistable p) throws StorageFullException {
+        Connection c;
+        try {
+            c = getConnection();
+            SqlHelper.updateToTable(c, tableName, p, id);
+            c.close();
+        } catch (SQLException | ClassNotFoundException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void remove(int id) {
+
+    }
+
+    @Override
+    public void remove(Persistable p) {
+        this.read(p.getID());
+    }
+
+    @Override
+    public void removeAll() {
+
+    }
+
+    @Override
+    public Vector<Integer> removeAll(EntityFilter ef) {
+        return null;
+    }
+
+    @Override
+    public void repack() {
+        //Unecessary!
+    }
+
+    @Override
+    public void repair() {
+        //Unecessary!
+    }
+
+    @Override
+    public void setReadOnly() {
+        //TODO: This should have a clear contract.
+    }
+
+    @Override
+    public void registerIndex(String filterIndex) {
+        // TODO Auto-generated method stub
+    }
+
+    @Override
+    public Iterator<T> iterator() {
+        return iterate();
+    }
+
+    private void tryCreateTable() {
+        try {
+            Connection c = getConnection();
+            SqlHelper.createTable(c, tableName, prototype.newInstance());
+            c.close();
+        } catch (SQLException | InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+            System.out.println("Couldn't create table: " + tableName + " got: " + e);
```

Comment:
Assuming that you expect the SQLexception to be the thing that is capable of failing here because there is already a table, it is the _only_ thing that should be swallowing the error, the other exceptions here are still meaningful